.circleci/config.yml

```diff
@@ -45,11 +45,55 @@ jobs:
       - run:
           name: Run smoketests
           command: |
-            docker run -it -e RUNNING_IN_ARTMAN_DOCKER=True --name smoketest artman /bin/bash -c "git clone https://github.com/googleapis/googleapis /tmp/googleapis_smoketest; python3 /artman/test/smoketest_artman.py --root-dir=/tmp/googleapis_smoketest --log=/tmp/smoketest.log" || OUT=$?
-            docker cp smoketest:/tmp/smoketest.log /tmp/smoketest.log
-            if [ $OUT -ne 0 ];then
-              exit $OUT
+            if [ "${RUN_SMOKETEST}" ]; then
+              docker run -it -e RUNNING_IN_ARTMAN_DOCKER=True --name smoketest artman /bin/bash -c "git clone https://github.com/googleapis/googleapis /tmp/googleapis_smoketest; python3 /artman/test/smoketest_artman.py --root-dir=/tmp/googleapis_smoketest --log=/tmp/smoketest.log" || OUT=$?
+              docker cp smoketest:/tmp/smoketest.log /tmp/smoketest.log
+              if [ $OUT -ne 0 ];then
+                exit $OUT
+              fi
+            else
+              echo "Environment variables RUN_SMOKETEST is not properly set in CircleCI project."
+              echo "Skip smoketest step."
             fi
+      - store_artifacts:
+          path: /artman/test/golden/actual_library_example.golden
+      - store_artifacts:
+          path: /tmp/smoketest.log
+
+  build_and_release_docker_image:
+    working_directory: /usr/src/artman
+    docker:
+      - image: docker:17.05.0-ce-git
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run:
+          name: Fingers crossed!
+          command: |
+            echo "Fingers crossed!"
+      - restore_cache:
```

Comment:
Is there a way to factor out this set-up? If unknown, a TODO?

Reply:
`restore_cache` is a circleci built-in task. What do you mean by factor out?

Reply:
I mean that lines 74-96 duplicate lines 15-37.